go/feast/featurestore.go

```diff
@@ -0,0 +1,907 @@
+		if featureService, ok = parsedKind.(*FeatureService); !ok {
+			return nil, errors.New("Cannot parse FeatureService from request")
+		}
+
+		var featureViewName string
+		features = make([]string, 0)
+		for _, featureProjection := range featureService.projections {
+			featureViewName = featureProjection.nameToUse()
+			for _, feature := range featureProjection.features {
+				features = append(features, fmt.Sprintf("%s:%s", featureViewName, feature.name))
+			}
+		}
+		return features, nil
+	} else {
+		return features, nil
+	}
+}
+
+/*
+	If features passed into GetOnlineFeaturesRequest as a list of feature references,
+		return all FeatureView, OnDemandFeatureView, RequestFeatureView from the registry
+	Otherwise, a FeatureService was passed, return a list of copies of FeatureViewProjection
+		copied from FeatureView, OnDemandFeatureView, RequestFeatureView existed in the registry
+
+	TODO (Ly): Since the implementation of registry has changed, a better approach here is just
+		retrieving featureViews asked in the passed in list of feature references instead of
+		retrieving all feature views. Similar argument to FeatureService applies.
+
+*/
+func (fs *FeatureStore) getFeatureViewsToUse(parsedKind interface{}, allowCache, hideDummyEntity bool) ([]*FeatureView, []*RequestFeatureView, []*OnDemandFeatureView, error) {
+
+	fvs := make(map[string]*FeatureView)
+	requestFvs := make(map[string]*RequestFeatureView)
+	odFvs := make(map[string]*OnDemandFeatureView)
+
+	featureViews, err := fs.listFeatureViews(allowCache, hideDummyEntity)
+	if err != nil {
+		// TODO (Ly): Review: return an empty list instead of
+		// error if no FeatureView is found in this project
+		featureViews = make([]*FeatureView, 0)
+	}
+	for _, featureView := range featureViews {
+		fvs[featureView.base.name] = featureView
+	}
+
+	requestFeatureViews, err := fs.registry.listRequestFeatureViews(fs.config.Project)
+	if err != nil {
+		// TODO (Ly): Review: return an empty list instead of
+		// error if no RequestFeatureView is found in this project
+		requestFeatureViews = make([]*RequestFeatureView, 0)
+	}
+	for _, requestFeatureView := range requestFeatureViews {
+		requestFvs[requestFeatureView.base.name] = requestFeatureView
+	}
+
+	onDemandFeatureViews, err := fs.registry.listOnDemandFeatureViews(fs.config.Project)
+	if err != nil {
+		// TODO (Ly): Review: return an empty list instead of
+		// error if no OnDemandFeatureView is found in this project
+		onDemandFeatureViews = make([]*OnDemandFeatureView, 0)
+	}
+	for _, onDemandFeatureView := range onDemandFeatureViews {
+		odFvs[onDemandFeatureView.base.name] = onDemandFeatureView
+	}
+
+	if featureService, ok := parsedKind.(*FeatureService); ok {
+
+		fvsToUse := make([]*FeatureView, 0)
+		requestFvsToUse := make([]*RequestFeatureView, 0)
+		odFvsToUse := make([]*OnDemandFeatureView, 0)
+
+		for _, featureProjection := range featureService.projections {
+			// Create copies of FeatureView that may
```

Comment:
Nitpick: I think we can make these comments a little bit longer before breaking them. Is there some kind of length you are shooting for?